### ISSUE TITLE
Error "Unspecified reason" when using Google Chrome 71.x+ for WebSocket transport

### ISSUE THREAD

**Issue by joa** [improbable-eng/grpc-web#377]
**Versions of relevant software used**
* Client `https://registry.yarnpkg.com/@improbable-eng/grpc-web/-/grpc-web-0.9.1.tgz#61a36fb04f6f2a202fe01b3a35d1fc891cc93933`
* Server `github.com/improbable-eng/grpc-web v0.9.1` (previously we used `github.com/improbable-eng/grpc-web v0.0.0-20181128220516-1c63d72d1d2f` and it didn't work either)

**What happened**
> WebSocket connection to 'wss://api.com/foo.TokenService/CreateToken' failed: Unspecified reason

**What you expected to happen**
No failure.

This seems like a regression and might be related to either an update of Google Chrome and/or the grpc-web client making assumptions about how a browser does WS when it shouldn't.

With Chrome 70.x, the CreateToken request is 117b, with 74.x it is only 57b in size.

**How to reproduce it (as minimally and precisely as possible)**:
* ✅ Works with Google Chrome 70.0.3538.77 (Official Build) (64-Bit)
* ✅ Works with Google Chrome 71.0.3578.30-1 (Official Build) beta (64-Bit)
* ❌ Fails with Google Chrome 74.0.3729.28 (Official Build) beta (64-bit)

**Additional Information**
We can see in the logs of our backend services that the actual request is coming through and being answered successfully. However the client seems to err out before. Approximately 1 out of 10 times this even works so it is not a consistent failure.

The code errs out because there are no responseHeaders present when the WS connection is closed.

I am able to run any tests or provide additional information that might help to resolve this.

Messages for a failed request:
![image](https://user-images.githubusercontent.com/141269/54911886-b3ab2980-4eef-11e9-8a20-26eb45d4a01d.png)

Messages for a successful request:
![image](https://user-images.githubusercontent.com/141269/54911944-cf163480-4eef-11e9-946d-71bf17a649d6.png)

**Comment by joa** [comment]
FYI as a workaround I can switch to `FetchReadableStreamTransport` for now. It works without any of the aforementioned issues.

**Comment by jonny-improbable** [comment]
Hi @joa thanks for raising.

As you point out above this could simply be a bug/quirk of an upcoming Chrome beta, or could be a breaking change in their handling of websockets which we are about to run into.

I don't currently have the bandwidth to investigate, however we do rely on the WebSocket transport internally at Improbable so I'll get one of the team to look into this when they get back from holiday next week.

**Comment by joa** [comment]
Thank you @jonny-improbable -- indeed, this is an issue we should also raise with the Chrome team as well. Worst case is this will hit the stable channel before a solution was found.

**Comment by jonny-improbable** [comment]
Hey @joa, we've been able to reproduce this internally and are setting up a repro-case outside of grpc-web to narrow down if this is a Chrome issue, or a grpc-web issue. At present we are leaning towards the former, in which case we will raise an upstream issue.

**Comment by jonny-improbable** [comment]
[All Chrome WebSocket issues](https://bugs.chromium.org/p/chromium/issues/list?can=2&q=os%3Dchrome+component%3ABlink%3ENetwork%3EWebSockets+)

**Comment by MarcusLongmuir** [comment]
I reproduced the issue using a Gorilla Websockets example - the library that grpc-web uses for the Websocket transport.

The issue arises when the websocket is closed immediately after opening. Sending messages isn't required. Delaying before closing appears to avoid the issue.

This is the gist of the reproduction: https://gist.github.com/MarcusLongmuir/c097d270c9b8c287cda6abbc53f0b472

As @jonny-improbable identified - this is the bug: [Issue 937790](https://bugs.chromium.org/p/chromium/issues/detail?id=937790). The bug appears to have been fixed by [this commit](https://chromium.googlesource.com/chromium/src.git/+/01b3aaef2154549f921a3b1d907ef688f2d90c8b).

I've tested in the latest Chromium release (`Version 75.0.3757.0`) and the issue is no longer present.
